problem_builder/step.py

```diff
@@ -75,40 +83,109 @@ class StepMixin(object):
         scope=Scope.content
     )
 
+    editable_fields = ('display_name', 'show_title',)
+
+    def _(self, text):
+        """ Translate text """
+        return self.runtime.service(self, "i18n").ugettext(text)
+
     @lazy
-    def step_number(self):
-        return list(self.get_parent().steps).index(_normalize_id(self.scope_ids.usage_id)) + 1
+    def siblings(self):
+        return self.get_parent().steps
 
     @lazy
-    def lonely_step(self):
-        if _normalize_id(self.scope_ids.usage_id) not in self.get_parent().steps:
-            raise ValueError("Step's parent should contain Step", self, self.get_parent().steps)
-        return len(self.get_parent().steps) == 1
+    def child_number(self):
+        return list(self.siblings).index(_normalize_id(self.scope_ids.usage_id)) + 1
+
+    @lazy
+    def lonely_child(self):
+        if _normalize_id(self.scope_ids.usage_id) not in self.siblings:
+            message = u"{child_caption}'s parent should contain {child_caption}".format(child_caption=self.CAPTION)
+            raise ValueError(message, self, self.siblings)
+        return len(self.siblings) == 1
 
     @property
     def display_name_with_default(self):
         """ Get the title/display_name of this question. """
         if self.display_name:
             return self.display_name
-        if not self.lonely_step:
-            return self._(u"Question {number}").format(number=self.step_number)
-        return self._(u"Question")
+        if not self.lonely_child:
+            return self._(u"{child_caption} {number}").format(
+                child_caption=self.CAPTION, number=self.child_number
+            )
+        return self._(self.CAPTION)
 
-    def author_view(self, context):
-        context = context.copy() if context else {}
-        context['hide_header'] = True
-        return self.mentoring_view(context)
+    @property
+    def allowed_nested_blocks(self):
+        """
+        Returns a list of allowed nested XBlocks. Each item can be either
+        * An XBlock class
+        * A NestedXBlockSpec
+
+        If XBlock class is used it is assumed that this XBlock is enabled and allows multiple instances.
+        NestedXBlockSpec allows explicitly setting disabled/enabled state, disabled reason (if any) and single/multiple
+        instances
+        """
+        return [
+            NestedXBlockSpec(AnswerBlock, boilerplate='studio_default'),
+            MCQBlock, RatingBlock, MRQBlock, HtmlBlockShim,
+            AnswerRecapBlock, MentoringTableBlock,
+        ]
 
-    def author_preview_view(self, context):
-        context = context.copy() if context else {}
-        context['hide_header'] = True
-        return self.student_view(context)
+    @property
+    def steps(self):
+        """ Get the usage_ids of all of this XBlock's children that are "Questions" """
```

Comment:
@itsjeyd they are questions, yes, but searched by `StepMixin` - a bit confusing. Supports suggestion to rename StepMixin to QuestionMixin